### DISCUSSION TITLE
PostgreSQL Passwordless Connection - Ubuntu

### DISCUSSION THREAD

**Discussion by xdaruis** [prisma/prisma#25032] [Q&A]
### Question

I am trying to do a passwordless authentication to my PSQL database. It's working perfectly on my local machine, on MacOS, but when deploying it the server on Ubuntu I get the following error.

I have other applications doing passwordless authentication to PSQL but with prisma I keep getting errors.

```
Error: P1000: Authentication failed against database server at localhost, the provided database credentials for ubuntu are not valid.
```

Any ideas or things I could try and fix this without setting a password auth?

### Information about Prisma Schema, Client Queries and Environment (optional)

prisma.schema
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

.env
```
DATABASE_URL="postgresql://ubuntu@localhost:5432/mydb?schema=public"
```

prisma version
```
"@prisma/client": "^5.18.0",
"prisma": "^5.18.0",
```

**Comment by nurul3101** [comment]
Hey @xdaruis 👋 

Are you able to connect to this database via a GUI tool like PgAdmin, TablePlus or DataGrip?

I assume you have configured your `pg_hba.conf` file for enabling passwordless authentication, right?

Is there any specific reason you don't want to have a password. Having a strong password is generally recommended for almost all cases.

**Reply by xdaruis** [reply]
Hey @nurul3101,👋

Thanks for the fast reply! 😎

"I assume you have configured your pg_hba.conf file for enabling passwordless authentication, right?" -> Yes. I can also connect through the terminal using `psql mydb` so I don't think that is the issue.

**Reply by ludralph** [reply]
Hi @xdaruis 

Do you still need help resolving this issue? From the [code](https://github.com/prisma/prisma/blob/531cb879ddcc8889a7320b5e55a120988d7bec3d/packages/internals/src/convertCredentials.ts#L82-L123), it looks like you should be able to connect.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
